content_scripts/link_hints.coffee

```diff
@@ -673,5 +674,25 @@ class TypingProtector extends Mode
 
     @onExit callback
 
+# TODO(mrmr1993): Use event.deepPath to traverse into shadow DOMs, as part of a solution to #1861. This
```

Comment:
We have append target node's ancestor to `registrationElement` if `document.contains` returns `false`, so I think, there won't be shadow DOMs on the path from listener to real target node. So what about you?